src/picomsg/src/oper_db_writer.cpp

```diff
@@ -114,3 +142,249 @@ bool oper_db_writer_set_terminator_vrf_tunnel_counter(const uint32_t terminator_
+  	corm_remote_public_address.field_id = LIBOPER_DB_CLIENT_IPV4_ADDRESS;
+	corm_local_network.str = oper_data.local_network;
+	corm_local_network.field_id = LIBOPER_DB_CLIENT_IPV4_ADDRESS;
+
+	if (! db_obj->union_set(db_obj.get(), LIBOPER_DB_CLIENT_FIELD_ID_TUNNEL_OPER_DATA_DEVICE_PUBLIC_ADDRESS, &corm_device_public_address)) {
+		PICOMSG_LOG(PICOMSG_LOG_ERR, "failed to set oper DB object LIBOPER_DB_CLIENT_FIELD_ID_TUNNEL_OPER_DATA_DEVICE_PUBLIC_ADDRESS to %s",
+					oper_data.device_public_address);
+		return false;
+	}
+	if (! db_obj->union_set(db_obj.get(), LIBOPER_DB_CLIENT_FIELD_ID_TUNNEL_OPER_DATA_REMOTE_PUBLIC_ADDRESS, &corm_remote_public_address)) {
+		PICOMSG_LOG(PICOMSG_LOG_ERR, "failed to set oper DB object LIBOPER_DB_CLIENT_FIELD_ID_TUNNEL_OPER_DATA_REMOTE_PUBLIC_ADDRESS to %s",
+					oper_data.remote_public_address);
+		return false;
+	}
+	if (! db_obj->union_set(db_obj.get(), LIBOPER_DB_CLIENT_FIELD_ID_TUNNEL_OPER_DATA_LOCAL_NETWORK, &corm_local_network)) {
+		PICOMSG_LOG(PICOMSG_LOG_ERR, "failed to set oper DB object LIBOPER_DB_CLIENT_FIELD_ID_TUNNEL_OPER_DATA_LOCAL_SRC_NETWORK to %s",
+					oper_data.local_network);
+		return false;
+	}
+
+	if (! db_obj->u16_set(db_obj.get(), LIBOPER_DB_CLIENT_FIELD_ID_TUNNEL_OPER_DATA_DEVICE_PUBLIC_SRC_PORT, oper_data.device_public_src_port)) {
+		PICOMSG_LOG(PICOMSG_LOG_ERR, "failed to set oper DB object enum LIBOPER_DB_CLIENT_FIELD_ID_TUNNEL_OPER_DATA_DEVICE_PUBLIC_SRC_PORT to %d",
+					oper_data.device_public_src_port);
+		return false;
+	}
+
+	if(write_uptime) {
+		if (! db_obj->timeticks_set_curr_timestamp(db_obj.get(), LIBOPER_DB_CLIENT_FIELD_ID_TUNNEL_OPER_DATA_TUNNEL_UPTIME)) {
+			PICOMSG_LOG(PICOMSG_LOG_ERR, "failed to set oper DB object enum LIBOPER_DB_CLIENT_FIELD_ID_TUNNEL_OPER_DATA_TUNNEL_UPTIME");
+			return false;
+		}
+	}
+
+	if (! (*oper_db_client)->set((*oper_db_client).get(), db_obj.release(), 0)) {
+		PICOMSG_LOG(PICOMSG_LOG_ERR, "failed to write vrf[%s] device-id[%s] tunnel oper data to oper DB", oper_data.vrf_name, oper_data.device_id);
+		return false;
+	}
+
+	return true;
+}
+
+bool oper_db_writer_remove_tunnel_oper_data(const char *vrf_name, const char *device_id) {
+	auto oper_db_client = picomsg_get_oper_db_client();
+	const char* keys[] = {vrf_name, device_id};
+
+	auto db_obj = oper_db_client_create_db_object(*oper_db_client, LIBOPER_DB_CLIENT_CLASS_ID_TUNNEL_ITEM, keys);
+
+	if (! (*oper_db_client)->del((*oper_db_client).get(), db_obj.release(), 0)) {
+		PICOMSG_LOG(PICOMSG_LOG_ERR, "failed to delete LIBOPER_DB_CLIENT_CLASS_ID_TUNNEL_ITEM keys {%s,%s} from oper DB",
+		vrf_name, device_id);
+		return false;
+	}
+
+	db_obj = oper_db_client_create_db_object(*oper_db_client, LIBOPER_DB_CLIENT_CLASS_ID_TUNNEL_OPER_DATA, keys);
+	if (! (*oper_db_client)->del((*oper_db_client).get(), db_obj.release(), 0)) {
+		PICOMSG_LOG(PICOMSG_LOG_ERR, "failed to delete LIBOPER_DB_CLIENT_CLASS_ID_TUNNEL_OPER_DATA keys {%s,%s} from oper DB",
+		keys[0], keys[1]);
```

Comment:
Pass `vrf_name, device_id`, to be aligned with the convention with the rest of the code (including this function itself)